Dockerfile

```diff
@@ -47,18 +42,17 @@ RUN apk add --no-cache \
       jq \
       yq
 
+
 COPY requirements.txt /opt/mautrix-telegram/requirements.txt
+COPY optional-requirements.txt /opt/mautrix-telegram/optional-requirements.txt
 WORKDIR /opt/mautrix-telegram
+RUN apk add --virtual .build-deps python3-dev libffi-dev build-base \
+ && pip3 install -r requirements.txt -r optional-requirements.txt \
+ && apk del .build-deps
 
 COPY . /opt/mautrix-telegram
-RUN apk add git \
-  && apk add --virtual .build-deps python3-dev libffi-dev build-base \
-  && pip3 install .[speedups,hq_thumbnails,metrics,e2be] \
-  # TODO: unpin Pillow here after it's updated in Alpine
-  && pip3 install -r requirements.txt 'pillow==8.2' \
-  && pip3 install 'git+https://github.com/vector-im/mautrix-python@v0.11.4-mod-2#egg=mautrix' \
```

Comment:
I believe the changes we wanted in the upstream library got merged in the end, so I dropped this.